src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs

```diff
@@ -193,6 +200,92 @@ private List<string> ResolveByExternalReferencePackages(List<string> uidList, Co
             return list;
         }
 
+        private async Task<List<string>> ResolveByXRefServiceAsync(List<string> uidList, ConcurrentDictionary<string, XRefSpec> externalXRefSpec)
+        {
+            if (XRefServiceUrls == null || XRefServiceUrls.Length == 0)
+            {
+                Logger.LogWarning($"You haven't provide an xrefservice item in docfx.json or command options!");
+                return uidList;
+            }
+            string requestUrl = XRefServiceUrls[0];
```

Comment:
it's a question, not decide the final form at right now.

---
In reply to: [126851373](https://github.com/dotnet/docfx/pull/1837#discussion_r126851373) [](ancestors = 126851373)